src/test/app/PayChan_test.cpp

```diff
@@ -1136,6 +1152,101 @@ struct PayChan_test : public beast::unit_test::suite
+            env.fund (XRP (10000), alice, bob);
+            env(create(alice, bob, XRP(1000), settleDelay, pk));
+            auto const [chan, chanSle] = channelKeyAndSle(*env.current(), alice, bob);
+            BEAST_EXPECT(inOwnerDir(*env.current(), alice, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), alice) == 1);
+            BEAST_EXPECT(!inOwnerDir(*env.current(), bob, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), bob) == 0);
+            // close the channel
+            env(claim(bob, chan), txflags(tfClose));
+            BEAST_EXPECT(!channelExists(*env.current(), chan));
+            BEAST_EXPECT(!inOwnerDir(*env.current(), alice, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), alice) == 0);
+            BEAST_EXPECT(!inOwnerDir(*env.current(), bob, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), bob) == 0);
+        }
+
+        {
+            // Test with adding the paychan to the recipients owner directory
+            Env env(*this, supported_amendments());
+            env.fund (XRP (10000), alice, bob);
+            env(create(alice, bob, XRP(1000), settleDelay, pk));
+            auto const [chan, chanSle] = channelKeyAndSle(*env.current(), alice, bob);
+            BEAST_EXPECT(inOwnerDir(*env.current(), alice, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), alice) == 1);
+            BEAST_EXPECT(inOwnerDir(*env.current(), bob, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), bob) == 1);
+            // close the channel
+            env(claim(bob, chan), txflags(tfClose));
+            BEAST_EXPECT(!channelExists(*env.current(), chan));
+            BEAST_EXPECT(!inOwnerDir(*env.current(), alice, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), alice) == 0);
+            BEAST_EXPECT(!inOwnerDir(*env.current(), bob, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), bob) == 0);
+        }
+
+        {
+            // Test removing paychans created before adding to the recipient's
+            // owner directory
+            Env env(
+                *this, supported_amendments() - fixPayChanRecipientOwnerDir);
+            env.fund (XRP (10000), alice, bob);
+            // create the channels before the amendment activates
+            env(create(alice, bob, XRP(1000), settleDelay, pk));
+            auto const [chan, chanSle] = channelKeyAndSle(*env.current(), alice, bob);
+            BEAST_EXPECT(inOwnerDir(*env.current(), alice, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), alice) == 1);
+            BEAST_EXPECT(!inOwnerDir(*env.current(), bob, chanSle));
+            BEAST_EXPECT(ownerDirCount(*env.current(), bob) == 0);
+            env.enableFeature(fixPayChanRecipientOwnerDir);
+            env.close();
+            BEAST_EXPECT(
+                env.current()->rules().enabled(fixPayChanRecipientOwnerDir));
```

Comment:
After enabling, consider testing if bob now has in his owner directory and the count.
e.g.
```
            BEAST_EXPECT(inOwnerDir(*env.current(), bob, chanSle));
            BEAST_EXPECT(ownerDirCount(*env.current(), bob) == 1);
```

Reply:
So this is really interesting. It's intended that Bob's owner count should be zero and the channel is not part of the owner directory. However, as written, this isn't true. I need to `close` after creating the channel or the open ledger runs with one set of amendments and when closing runs with another set of amendments.

I fixed this in `499e9e0e3 Add `close` after channel create and fix typos`